Sources/DVR/Interaction.swift

```diff
@@ -2,21 +2,24 @@ import Foundation
 
 struct Interaction {
 
-    // MARK: - Properties
+    // MARK: - Propertie6s
 
-    let request: URLRequest
-    let response: Foundation.URLResponse
-    let responseData: Data?
+    var request: URLRequest
+    var response: Foundation.URLResponse
+    var responseData: Data?
     let recordedAt: Date
-
+    let filter: Filter?
 
     // MARK: - Initializers
 
-    init(request: URLRequest, response: Foundation.URLResponse, responseData: Data? = nil, recordedAt: Date = Date()) {
+    init(request: URLRequest, response: Foundation.URLResponse, responseData: Data? = nil, recordedAt: Date = Date(), filter: Filter? = nil) {
```

Comment:
It seems to me the filter should be associated with the session, not individual interactions.